src/module/actor/character/document.ts

```diff
@@ -1815,32 +1816,37 @@ class CharacterPF2e extends CreaturePF2e {
                     return "";
                 }
 
-                const damage = WeaponDamagePF2e.calculate({
+                const outcome = method === "damage" ? "success" : "criticalSuccess";
+                const { self, target, options } = context;
+                const damageContext: DamageRollContext = {
+                    type: "damage-roll",
+                    sourceType: "attack",
+                    self,
+                    target,
+                    outcome,
+                    options,
+                    domains,
+                };
+
+                const rollParams = params.event ? eventToRollParams(params.event) : {};
```

Comment:
Would there be any consequences here if eventToRollParams() allowed event to be optional, and purely used the settings in that situation?

Reply:
Probably not. I'll check it out.

Reply:
I could imagine you could ... it into the damage context, then you'd only need to tweak skipDialog specifically.